geniza/annotations/views.py

```diff
@@ -0,0 +1,90 @@
+import json
+
+from django.http import HttpResponse, JsonResponse
+from django.shortcuts import render
+from django.utils.decorators import method_decorator
+from django.views.decorators.csrf import csrf_exempt
+from django.views.decorators.http import require_http_methods
+from django.views.generic.base import View
+from django.views.generic.detail import SingleObjectMixin
+from django.views.generic.list import MultipleObjectMixin
+
+from geniza.annotations.models import Annotation
+
+
+class AnnotationResponse(JsonResponse):
+    content_type = 'application/ld+json; profile="http://www.w3.org/ns/anno.jsonld"'
+
+
+@method_decorator(csrf_exempt, name="dispatch")  # disable csrf for testing with curl
+class AnnotationList(View, MultipleObjectMixin):
+    model = Annotation
+    http_method_names = ["get", "post"]
+
+    paginate_by = None  # disable pagination for now
+
+    def get(self, request, *args, **kwargs):
+        # strictly speaking, annotations endpoint should return an annotation container,
+        # but that structure looks terrible to work with and we need a way to search,
+        # which is not defined in the w3c annotation protocol.
```

Comment:
Is that what the endpoint returns in SAS? And by searching you mean allowing the overall geniza search to search transcription content, right?

Reply:
SAS search endpoint just returns a list of annotations. Django doesn't like returning non-dicts by default, so I thought I'd wrap it in a dict, esp. since we may need to add pagination or other properties. 

The SAS annotation endpoint returns an AnnotationList, but that doesn't seem to exist in w3c annotation; we could model it on `AnnotationPage` or `AnnotationCollection`. Strictly speaking it should be an annotation collection; I didn't like the way the results are nested (`first.items`) but I see now that it's nested that way because the `first` item is an `AnnotationPage`.  Maybe that's not terrible.

Reply:
I see. Took a look at the W3C spec now. It does seem that's the preferred way to collect annotations into a list. I lean towards following the spec as closely as possible, but I also totally get the hesitation if it adds a bunch of overhead to this (as well as things like searching). I'll leave it up to you, but my vote would be for `AnnotationCollection`.

Reply:
I'm not opposed to supporting annotation collection at some point, but my goal at the moment is to get us a minimal working version that we can use with our existing annotation code and deliver transcription functionality for testing. We're not using the main endpoint at all, except to search for annotations by target uri so we can load them for display.

I looked again, I can't find anything about search anywhere in the w3c annotation protocol or model specs, or in the LDP doc (annotation container is an LDP). I'm biased against LDP because that's what Fedora Commons 4 was supposed to be built on, and my understanding is that it ended up being so inefficient that the Hydra/Samvera community basically abandoned it in favor of the postgres-based Valkyrie that the PUL devs created.

I did look briefly at iiif search API https://iiif.io/api/search/1.0/ — this is what SAS implements for search (separate from the base annotations endpoint), which explains why it returns an AnnotationList. I can see the value of that, actually, because an annotation list can be referenced in a iiif manifest.  This part does impact our current code, since we need to know where to search for annotations on load.  

What do you think about a separate search endpoint with a minimal implementation (for now) that returns an annotation list when you search by target uri?

Reply:
That makes sense to me! Good point about not using the main search endpoint at all yet—it doesn't seem needed at this point. Thank you for the detailed look into the options here.